data/kernels/denoiseprofile.cl

```diff
@@ -86,16 +86,26 @@ denoiseprofile_dist(read_only image2d_t in, global float* U4, const int width, c
   const int y = get_global_id(1);
   const int gidx = mad24(y, width, x);
 
-  // Reminder: q.x and q.y can be negative
   if(x >= width || y >= height) return;
-  if(x+q.x >= width || y+q.y >= height) return;
-  if(x+q.x < 0 || y+q.y < 0) return;
+
+  int xpq = x + q.x;
+  int ypq = y + q.y;
+  // Convert out of bounds indexes to 0
+  // Reminder: q.x and q.y can be negative
+  // the boolean conditions will be equal to
+  // 0 in case of out of bounds, and will be
+  // equal to 1 in the other case
+  xpq *= (x+q.x < width && x+q.x >= 0);
+  ypq *= (y+q.y < height && y+q.y >= 0);
 
   float4 p1 = read_imagef(in, sampleri, (int2)(x, y));
-  float4 p2 = read_imagef(in, sampleri, (int2)(x, y) + q);
+  float4 p2 = read_imagef(in, sampleri, (int2)(xpq, ypq));
   float4 tmp = (p1 - p2)*(p1 - p2);
   float dist = tmp.x + tmp.y + tmp.z;
-  
+
+  // make dist equal to 0 in case xpq or ypq is out of bounds
+  dist *= (x+q.x < width && x+q.x >= 0 && y+q.y < height && y+q.y >= 0);
```

Comment:
Oops, pay attention: in OpenCL the logical TRUE value is -1, not +1 as expected from C programmers.

You will need to do something like:

dist *= (logical term) ? 1.0f : 0.0f;

Reply:
Oh, ok, thank you, I will change that.
Thanks for the review :-)